client/components/data/document-head/index.jsx

```diff
@@ -45,7 +45,10 @@ class DocumentHead extends Component {
 	}
 
 	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( nextProps.title !== undefined && this.props.title !== nextProps.title ) {
+		if (
+			nextProps.title !== undefined &&
+			this.props.title?.toString?.() !== nextProps.title?.toString?.()
```

Comment:
This does the job well, as `document.title` set as the translate React component will yield the translated string 👍  

Care to add a comment that describes why this additional check is necessary?